client/main.lua

```diff
@@ -161,36 +177,105 @@ local function busGarage()
     lib.showContext('open_busGarage_context_menu')
 end
 
+local function busMain()
+    if PlayerData.job.name == "bus" then
+        if not VehicleZone then
+            local inRange = false
+
+            VehicleZone = lib.zones.sphere({
+                name = "busMain",
+                coords = vec3(Config.Location.x, Config.Location.y, Config.Location.z),
+                radius = 5,
+                debug = false,
+                onEnter = function()
+                    inRange = true
+
+                    CreateThread(function()
+                        repeat
+                            Wait(0)
+
+                            local inVeh = whitelistedVehicle()
+
+                            if not inVeh then
+                                lib.showTextUI(Lang:t('info.busstop_text'))
+
+                                if IsControlJustReleased(0, 38) then
+                                    busGarage()
+
+                                    lib.hideTextUI()
+                                    break
+                                end
+                            else
+                                lib.showTextUI(Lang:t('info.bus_stop_work'))
+
+                                if IsControlJustReleased(0, 38) then
+                                    if not NpcData.Active or NpcData.Active and not NpcData.NpcTaken then
+                                        if IsPedInAnyVehicle(cache.ped, false) then
+                                            BusData.Active = false
+
+                                            DeleteVehicle(cache.vehicle)
+                                            RemoveBlip(NpcData.NpcBlip)
+
+                                            lib.hideTextUI()
+
+                                            resetNpcTask()
+                                            break
+                                        end
+                                    else
+                                        QBCore.Functions.Notify(Lang:t('error.drop_off_passengers'), 'error')
+                                    end
+                                end
+                            end
+                        until not inRange
+                    end)
+                end,
+                onExit = function()
+                    lib.hideTextUI()
+
+                    inRange = false
+                end
+            })
+        end
+    else
+        if VehicleZone then
+            VehicleZone:remove()
+        end
+    end
+end
```

Comment:
```suggestion
local function busMain()
    if PlayerData.job.name ~= "bus" then
        if VehicleZone then
            VehicleZone:remove()
        end
        return
    end

    if VehicleZone then return end
    local inRange = false

    VehicleZone = lib.zones.sphere({
        name = "busMain",
        coords = vec3(Config.Location.x, Config.Location.y, Config.Location.z),
        radius = 5,
         debug = false,
        onEnter = function()
            inRange = true

            CreateThread(function()
                repeat
                    Wait(0)

                    local inVeh = whitelistedVehicle()

                    if inVeh then
                        lib.showTextUI(Lang:t('info.bus_stop_work'))

                        if IsControlJustReleased(0, 38) then
                            if not NpcData.Active or NpcData.Active and not NpcData.NpcTaken then
                                if IsPedInAnyVehicle(cache.ped, false) then
                                    BusData.Active = false

                                    DeleteVehicle(cache.vehicle)
                                    RemoveBlip(NpcData.NpcBlip)

                                    lib.hideTextUI()

                                    resetNpcTask()
                                    break
                                end
                            else
                                QBCore.Functions.Notify(Lang:t('error.drop_off_passengers'), 'error')
                            end
                        end
                    else
                        lib.showTextUI(Lang:t('info.busstop_text'))

                        if IsControlJustReleased(0, 38) then
                            busGarage()

                            lib.hideTextUI()
                            break
                        end
                    end
                until not inRange
                end)
            end,
        onExit = function()
            lib.hideTextUI()
            inRange = false
        end
    })
end
```
Organising this function as such will improve readability and save on indentation